stan/math/rev/fun/generalized_inverse.hpp

```diff
@@ -66,8 +69,20 @@ inline auto generalized_inverse(const VarMat& G) {
   if (G.size() == 0)
     return ret_type(G);
 
-  if (G.rows() == G.cols())
-    return ret_type(inverse(G));
+  if (G.rows() == G.cols()) {
```

Comment:
For if's where only one branch should happen it's nicer to use `if else` after the first if